### PR TITLE
Fixes for the HRL Trainer

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

Bug fixes for the HRL trainer to properly work

## How Has This Been Tested

Training the HL policy on the rearrange easy task. 

## Types of changes

- **\[Bug Fix\]** (non-breaking change which fixes an issue)

### CODE REVIEW

File: habitat-lab/habitat/tasks/rearrange/utils.py [hunk 1]
```diff
@@ -412,6 +414,9 @@ def get_robot_spawns(
     """
 
     state = sim.capture_state()
+    if agent is None:
+        agent = sim.articulated_agent
+    targ_island_radius = sim.pathfinder.island_radius(target_position)
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/rearrange/utils.py:419]: Why use the radius? Wouldn't it be easier to access the navmesh island index ?

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 2]
```diff
@@ -303,6 +298,7 @@ def reconfigure(self, config: "DictConfig", ep_info: RearrangeEpisode):
             for handle, ro in rom.get_objects_by_handle_substring().items()
         }
 
+        rom = self.get_rigid_object_manager()
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:301]: duplicate with line 306

File: habitat-lab/habitat/tasks/rearrange/utils.py [hunk 3]
```diff
@@ -394,6 +395,7 @@ def get_robot_spawns(
     sim,
     num_spawn_attempts: int,
     physics_stability_steps: int,
+    agent: Optional[MobileManipulator] = None,
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/rearrange/utils.py:398]: missing docstring

**Review by vincentpierre — APPROVED** [review]
Some comments. Please address especially on the habitat-lab side.

File: habitat-lab/habitat/tasks/rearrange/utils.py [hunk 4]
```diff
@@ -437,8 +442,12 @@ def get_robot_spawns(
         if target_distance > distance_threshold or not is_navigable:
             continue
 
-        sim.articulated_agent.base_pos = start_position
-        sim.articulated_agent.base_rot = start_rotation
+        island_radius = sim.pathfinder.island_radius(start_position)
+        if island_radius != targ_island_radius:
+            continue
```
[the remark this answers is not in the document]
Comment (ASzot) [reply habitat-lab/habitat/tasks/rearrange/utils.py:447]: Agreed, let me delay this change for a later PR to get the right solution.

**Review by vincentpierre — APPROVED** [review]

**Pull request merged**